File: data-service/spring-data-provider/src/main/java/org/aykhan/dataprovider/dto/post/PostResponse.java
package org.aykhan.dataprovider.dto.post;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.aykhan.dataprovider.dto.comments.CommentResponse;
import org.aykhan.dataprovider.dto.user.UserResponse;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PostResponse {
  private Long id;
  private String title;
  private String body;
  private UserResponse author;
  private List<CommentResponse> comments;
  @JsonFormat(pattern = "HH:mm dd/MM/yyyy")
  private LocalDateTime updatedOn;
}

File: data-service/scala-sql-dataprovider/app/dtos/post/PostResponse.scala
package dtos.post

import java.time.LocalDateTime

import dtos.comment.CommentResponse
import dtos.user.UserResponseDTO
import play.api.libs.json.Json

case class PostResponse(id: Long,
                        title: String,
                        body: String,
                        author: Option[UserResponseDTO],
                        updatedOn: Option[LocalDateTime],
                        comments: List[CommentResponse])

object PostResponse {
  implicit val writes = Json.writes[PostResponse]
}
